### PR TITLE
Add ungrouped option to metrics

### PR DESCRIPTION
**Pull request by michielderoos**
**Describe the Pull Request**

Before this, metrics were always part of a group. No more!

**Todo**

- [x] Link relevant [trello card](https://trello.com/b/PA2LhlOh/sempo-dev) or [related issue](https://github.com/teamsempo/SempoBlockchain/issues) to the pull request
- [x] Request review from relevant @person or team
- [x] QA your pull request. This includes running the app, login, testing changes etc.
- [x] Bump [backend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/config.py#L7) and/or [frontend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/app/package.json#L3) versions following Semver

### CODE REVIEW

**Comment by michielderoos** [comment]
[Add a simple ungrouped "Total" to the Time Series charts, and base the % changes over a time period always off the first and last dates in this data.](https://trello.com/c/ZG0q9xTN/512-add-a-simple-ungrouped-total-to-the-time-series-charts-and-base-the-changes-over-a-time-period-always-off-the-first-and-last-dat)

File: app/server/utils/metrics/metric.py [hunk 1]
```diff
@@ -17,14 +17,18 @@ def execute_query(self, user_filters: dict = None, date_filters_dict=None, enabl
              aggregated_query and total_query
         """
         actions = { 'query': self.query_actions, 'aggregated_query': self.aggregated_query_actions, 'total_query': self.total_query_actions }
+
+        # Build the dict of queries to execute. Ungrouped metrics don't have aggregated queries,
+        # and sometimes we only want aggregates and totals (based on calculate_only_aggregates)
         if self.is_timeseries:
             if calculate_only_aggregates:
```
Comment (enjeyw) [inline app/server/utils/metrics/metric.py:24]: should this be `dont_include_timeseries` given that we now have `if self.aggregated_query:` below?

File: app/server/utils/metrics/metric.py [hunk 2]
```diff
@@ -59,13 +63,16 @@ def execute_query(self, user_filters: dict = None, date_filters_dict=None, enabl
                 results[query] = result
             else:
                 results[query] = postprocessing_actions.execute_postprocessing(result, population_query_result, actions[query])
-        
+
         if self.is_timeseries:
             result = {}
             if not calculate_only_aggregates:
```
Comment (enjeyw) [inline app/server/utils/metrics/metric.py:69]: same

File: app/server/utils/metrics/participant_stats.py [hunk 3]
```diff
@@ -57,16 +62,21 @@ def __init__(self, group_strategy, timeseries_unit = 'day'):
...
             total_query_actions=[GET_FIRST],
         ))
-    
-        active_users_timeseries_query = db.session.query(func.count(func.distinct(CreditTransfer.sender_user_id)).label('volume'),
-                func.date_trunc(self.timeseries_unit, CreditTransfer.created).label('date'), group_strategy.group_by_column).group_by(func.date_trunc(self.timeseries_unit, CreditTransfer.created))
-        aggregated_active_users_query = db.session.query(func.count(func.distinct(CreditTransfer.sender_user_id)).label('volume'), group_strategy.group_by_column)
+
+        if group_strategy:
+            active_users_timeseries_query = group_strategy.build_query_group_by_with_join(db.session.query(func.count(func.distinct(CreditTransfer.sender_user_id)).label('volume'),
+                    func.date_trunc(self.timeseries_unit, CreditTransfer.created).label('date'), group_strategy.group_by_column).group_by(func.date_trunc(self.timeseries_unit, CreditTransfer.created)), CreditTransfer)
```
Comment (enjeyw) [inline app/server/utils/metrics/participant_stats.py:68]: could consider turning `db.session.query(func.count(func.distinct(CreditTransfer.sender_user_id)).label('volume'), func.date_trunc(self.timeseries_unit, CreditTransfer.created).label('date')`  etc into vars to DRY up a little more, but NBD either way

**Pull request merged**